src/main/java/edu/bu/metcs673/trackr/controller/TransactionController.java

```diff
@@ -0,0 +1,90 @@
+
+
+    @PostMapping
+    public ResponseEntity<Transaction> createTransaction(@Valid @RequestBody TransactionDTO transactionInput) {
+
+        // pull username from JWT token, find corresponding user record
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(transactionInput.getBankAccountId(), user.getId());
+
+        // create a new bank account record associated to the user making the API request.
+        Transaction transaction = transactionService.createTransaction(transactionInput, bankAccount);
+
+        return new ResponseEntity<Transaction>(transaction, HttpStatus.OK);
+    }
+
+    @GetMapping("/find")
+    public ResponseEntity<Transaction> findTransactionById(@RequestParam("transactionId") long transactionId, @RequestParam("bankAccountId") long bankAccountId) {
+        // pull username from JWT token
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(bankAccountId, user.getId());
+
+        Transaction transaction = transactionService.findTraByIdAndBankAccountId(transactionId, bankAccount.getId());
+        return new ResponseEntity<Transaction>(transaction, HttpStatus.OK);
+
+    }
+
+    @GetMapping("/findAll")
+    public ResponseEntity<List<Transaction>> findTransactionById(@RequestParam("bankAccountId") long bankAccountId) {
+        // pull username from JWT token
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(bankAccountId, user.getId());
+
+        List<Transaction> transactions = transactionService.findAllTraByBankAccountId(bankAccount.getId());
+
+        return new ResponseEntity<List<Transaction>>(transactions, HttpStatus.OK);
+    }
+
+    @PostMapping("/modify")
+    public ResponseEntity<Transaction> modifyTransaction(@RequestParam("transactionId") long transactionId, @RequestBody TransactionDTO transactionInput) {
+        // pull username from JWT token
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(transactionInput.getBankAccountId(), user.getId());
+
+        Transaction transaction = transactionService.findTraByIdAndBankAccountId(transactionId, bankAccount.getId());
+
+        transactionService.modifyTransaction(transaction, transactionInput);
```

Comment:
i would modify this line to add `transaction = ...` to ensure that the transaction value is being reassigned

Reply:
OK, have done